hyperspy/_signals/eds.py

```diff
@@ -175,25 +177,61 @@ def sum(self, axis=None, out=None):
             return s
     sum.__doc__ = Signal1D.sum.__doc__
 
-    def rebin(self, new_shape, out=None):
-        new_shape_in_array = []
-        for axis in self.axes_manager._axes:
-            new_shape_in_array.append(
-                new_shape[axis.index_in_axes_manager])
-        factors = (np.array(self.data.shape) /
-                   np.array(new_shape_in_array))
-        s = super().rebin(new_shape, out=out)
-        s = out or s
-        # modify time per spectrum
-        this_md = self.metadata
-        that_md = s.metadata
-        keys = ("Acquisition_instrument.SEM.Detector.EDS.live_time",
-                "Acquisition_instrument.TEM.Detector.EDS.live_time")
-        for key in keys:
-            if key in this_md:
-                that_md.set_item(key, this_md.get_item(key) * np.prod(factors))
-        return s
-    sum.__doc__ = Signal1D.sum.__doc__
+    def rebin(self, new_shape=None, scale=None, crop=True, out=None):
+        """
+        %s
+        """
+
+    #Series of if statements to check that only one out of new_shape or scale
+    #has been given. New_shape is then converted to scale. If both or neither
+    #are given the function raises and error and wont run.
+
+        if new_shape == None and scale == None:
+            raise ValueError("One of new_shape, or scale must be specified")
+        elif new_shape != None and scale != None:
+            raise ValueError("Only one out of new_shape or scale should be specified.\
+                            Not both.")
+        elif new_shape != None:
+            scale = []
+            for i, axis in enumerate(self.data.shape):
+                scale.append(self.data.shape[i]/new_shape[i])
+        else:
+            new_shape = new_shape
+            scale = scale
```

Comment:
Actually this piece of code gets repeated at least 3-4 times, and thus should be moved to a separate function. I would move it to `array_tools` module or some place similar (i.e. not within a class at all), and its signature should be something along the lines of (feel free to use a better function name):
```python
new_shape, scale = _rebin_preprocess(new_shape, scale, data_shape)
```
Here the parameters should be whatever the user entered plus the actual current data shape, and the results should be what you'd have at this point in the function where this comment is. The new function should have the necessary corrections according to my previous comments